### PR TITLE
Add boolean data type support to Arrow Datasets

### LINKED ISSUE #78: Add boolean type support to Arrow Datasets
Arrow datasets are missing support for boolean data types

### PR DESCRIPTION
**Pull request by BryanCutler**
This change adds boolean data type support to Arrow based Datasets, both for scalar and vector tensors.  Additionally, checking Arrow batch column type matches the expected tensor output type has been moved from checking at each iteration, to only once per batch stream. Checking once per stream is all that is required since a set schema applies to all batches within that stream.

Fixes #78

### CODE REVIEW

**Comment by BryanCutler** [comment]
@yongtang @terrytangyuan please take a look when you have time, thank you!

File: tensorflow_io/arrow/python/kernel_tests/arrow_test.py [hunk 1]
```diff
@@ -305,6 +308,40 @@ def run_server(num_batches):
 
     server.join()
 
+  @unittest.skipIf(not _have_pyarrow, _pyarrow_requirement_message)
+  def testBoolArrayType(self):
+
+    # NOTE: need to test this seperately because to_pandas fails with
+    # ArrowNotImplementedError: Not implemented type for list in
```
Comment (xhochy) [inline tensorflow_io/arrow/python/kernel_tests/arrow_test.py]: Is this a known issue?

Comment (BryanCutler) [reply tensorflow_io/arrow/python/kernel_tests/arrow_test.py]: @xhochy  I'm not sure, I just noticed it with this pr. I'll look into it and make a jira if one doesn't exist already.

Comment (BryanCutler) [reply tensorflow_io/arrow/python/kernel_tests/arrow_test.py]: Found the issue https://issues.apache.org/jira/browse/ARROW-4370

Comment (xhochy) [reply tensorflow_io/arrow/python/kernel_tests/arrow_test.py]: @BryanCutler Thanks for digging into this!

Comment (terrytangyuan) [reply tensorflow_io/arrow/python/kernel_tests/arrow_test.py]: Could you add the link to the issue in the comment?

Comment (BryanCutler) [reply tensorflow_io/arrow/python/kernel_tests/arrow_test.py]: yup, good idea

**Comment by BryanCutler** [comment]
Thanks for reviewing @terrytangyuan !

**Pull request merged**